infra/serverless/lambda-web-app-test/main.tf

```diff
@@ -0,0 +1,220 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.16"
+    }
+  }
+
+  required_version = ">= 1.2.0"
+}
+
+provider "aws" {
+  region = "us-east-1"
+}
+
+#Lambda IAM Role
+resource "aws_iam_role" "iam_role_for_lambda" {
+  name = "iam_role_for_lambda"
+
+  assume_role_policy = <<EOS
+    {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+        "Action": "sts:AssumeRole",
+        "Principal": {
+            "Service": "lambda.amazonaws.com"
+        },
+        "Effect": "Allow",
+        "Sid": ""
+        }
+      ]
+    }
+  EOS
+}
+
+#Lambda Functon Setup
+resource "aws_lambda_function" "weather_app_lambda" {
+  filename      = "package.zip"
+  function_name = "weather_app_lambda"
+  role          = aws_iam_role.iam_role_for_lambda.arn
+  handler       = "index.handler"
+
+  source_code_hash = filebase64sha256("package.zip")

```

Comment:
Will probably want to update this if/when you bring the code for the package into this repository and have it as something thats build locally (and potentially under a different structure like `infra/zips/package.zip`)